adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```
@@ -0,0 +1,291 @@
+            while (true) {
+                try {
+                    queue.put(new String(cbuf, off, len));
+                    return;
+                } catch (InterruptedException ignored) {
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        @Override
+        public synchronized void flush() throws IOException {
+        }
+
+        @Override
+        public synchronized void close() throws IOException {
+            flush();
+            queue = null;
+        }
+
+    }
+
+    public class SocketAcceptor implements Runnable, Shutdown {
+        private final BlockingQueue<String> queue;
+        private final ServerSocket serverSocket;
+        private boolean running = true;
+
+        public SocketAcceptor(BlockingQueue<String> queue, ServerSocket serverSocket) {
+            this.queue = queue;
+            this.serverSocket = serverSocket;
+        }
+
+        public void shutdown() {
+            this.running = false;
+        }
+
+        @Override
+        public void run() {
+            try (ServerSocket serverSocket = this.serverSocket) {
+                while (running) {
+                    serverSocket.setSoTimeout(1000);
+                    serverSocket.setReuseAddress(true);
+                    try {
+                        Socket connectedSocket = serverSocket.accept();
+                        SocketWriter writer = new SocketWriter(queue, connectedSocket);
+                        //TCPServerActivity.this.managedShutdown.add(writer);
+                        Thread writerThread = new Thread(writer);
+                        writerThread.setName("SocketWriter/" + connectedSocket);
+                        writerThread.setDaemon(true);
+                        writerThread.start();
+                        logger.info("Started writer thread for " + connectedSocket);
+                    } catch (SocketTimeoutException ignored) {
```

I'm wondering for the cases in NB where we ignore exceptions if we should at least put in a logger.debug(...) entry to indicate that it happened in case we need to turn on debug level to diagnose a strange issue.  wdyt?

---

I think extra help is always appreciated when debugging, I added a logger debug message indicating the socket where the exception is thrown.